.github/workflows/test-go.yml

```diff
@@ -240,7 +240,9 @@ jobs:
           fi
 
           # On a release branch, add a flag to rerun failed tests
-          if [[ "${{ github.ref_name }}" = "release/*" ]]; then
+          # shellcheck disable=SC2193 # can get false positive for this comparision 
+          if [[ -n "${{ github.base_ref }}" && "${{ github.base_ref }}" == release/* ]] || [[ "${{ github.ref_name }}" == release/* ]]
```

Comment:
I think this is equivalent to

```
if [[  "${{ github.base_ref }}" == release/* ]] || [[ "${{ github.ref_name }}" == release/* ]]
```

But I think what we really want is more like:

```
if [[  "${{ github.base_ref }}" == release/* ]] || [[  -z "${{ github.base_ref }}" && "${{ github.ref_name }}" == release/* ]]
```